visualization/app/codeCharta/util/fileValidator.ts

```diff
@@ -134,3 +151,70 @@ function getAsApiVersion(version: string): ApiVersion {
 		minor: Number(version.split(".")[1])
 	}
 }
+
+function validateFixedFolders(file: ExportCCFile, result: CCValidationResult) {
+	const notFixed: string[] = []
+	const outOfBounds: string[] = []
+	const intersections: Set<string> = new Set()
+
+	file.nodes[0].children.forEach(node => {
+		if (node.fixed === undefined) {
+			notFixed.push(node.name)
+		} else {
+			if (isOutOfBounds(node)) {
+				outOfBounds.push(node.name)
+			}
+
+			file.nodes[0].children.forEach(node2 => {
+				if (node !== node2 && rectanglesIntersect(node.fixed, node2.fixed)) {
+					if (!intersections.has(`${node2.name} and ${node.name}`)) {
+						intersections.add(`${node.name} and ${node2.name}`)
+					}
```

Comment:
With the current implementation, we check for each corner point of `rect1` if is in `rect2`. We also do this the other way around. In order to prevent the same intersection being added twice, this check is required. Otherwise we might have intersections like: `["folder_1 and folder_2", "folder_2 and folder_1"]`

Reply:
I see 👍